pkg/cache/internal/informers_map.go

```diff
@@ -310,17 +334,17 @@ func (ip *InformersMap) addInformerToMap(gvk schema.GroupVersionKind, obj runtim
 	// Start the Informer if need by
```

Comment:
At the top of this function, shouldn't we check first if we're stopped already before getting all the way down here and return an error?

If I got the flow right, this function would still return correctly even if the informermap has been stopped

Reply:
Indeed, I still return correctly. I did this to make sure the behavior remained identical to before the change.

Reply:
It won't fail silently, it will error here (identical to current behavior): https://github.com/kubernetes-sigs/controller-runtime/blob/45e7aa7ab1fa4e8675848e2bcd9058c14cb0131a/pkg/cache/internal/informers_map.go#L266

I think that in order to prevent making a breaking change, we can keep the code as-is or throw the same timeout error from the addInformerToMap function.

Reply:
That makes sense, although it sounds to me it might be a user error to add after Stop has been called. Could we mark this as a breaking change and return that error?

Reply:
Ok, I'll need some more time to figure out how to reliably return an error. I agree that it would be useful to make a breaking change here.